src/engine/source/kvdb2/src/kvdbManager.cpp

```diff
@@ -67,100 +86,215 @@ void KVDBManager::initializeOptions()
+std::optional<base::Error> KVDBManager::deleteDB(const std::string& name)
+{
+    auto handlersInfo = getKVDBHandlersInfo();
 
-        m_cfHandles.push_back(cfHandle);
-        m_mapCFHandles.insert(std::make_pair(dbName, cfHandle));
+    if (handlersInfo.count(name))
+    {
+        return base::Error {fmt::format("Can not remove the DB. This is in use.")};
     }
 
-    result = m_kvdbHandlerCollection->getKVDBHandler(m_pRocksDB, cfHandle, dbName, scopeName);
+    auto it = m_mapCFHandles.find(name);
+    if (it != m_mapCFHandles.end())
+    {
+        auto cfHandle = it->second;
+        auto opStatus = m_pRocksDB->DestroyColumnFamilyHandle(cfHandle);
+        assert(opStatus.ok());
+        m_mapCFHandles.erase(it);
+    }
+    else
+    {
+        return base::Error {fmt::format("The DB not exists.")};
+    }
 
-    return result;
+    return std::nullopt;
 }
 
-void KVDBManager::removeKVDBHandler(const std::string& dbName, const std::string& scopeName)
+std::optional<base::Error> KVDBManager::createDB(const std::string& name)
 {
-    //TODO: check cfhandler is use for other scope
+    auto cfHandle = createColumnFamily(name);
 
-    if (m_mapCFHandles.find(dbName) != m_mapCFHandles.end())
+    if (nullptr != cfHandle)
     {
-        auto it = std::remove_if(m_cfHandles.begin(),
-                                 m_cfHandles.end(),
-                                 [&](auto const& cf)
-                                 { return cf->GetName() == dbName; });
-        m_cfHandles.erase(it);
-        m_kvdbHandlerCollection->removeKVDBHandler(dbName, scopeName);
+        return std::nullopt;
+    }
 
-        auto s = m_pRocksDB->DestroyColumnFamilyHandle(m_mapCFHandles[dbName]);
-        assert(s.ok());
+    return base::Error {fmt::format("Could not create DB.")};
+}
 
-        m_mapCFHandles.erase(dbName);
+std::optional<base::Error> KVDBManager::existsDB(const std::string& name)
+{
+    if (m_mapCFHandles.count(name))
+    {
+        return std::nullopt;
+    }
+
+    return base::Error {fmt::format("The DB not exists.")};
+}
+
+std::map<std::string, kvdbManager::RefInfo> KVDBManager::getKVDBScopesInfo()
+{
+    // List reverse lookup of getKVDBHandlersInfo. List of scopes and DBs that are using them.
+    std::map<std::string, kvdbManager::RefInfo> retValue;
+
+    // Retrieve the list of DBs and scopes that are using them.
+    std::map<std::string, kvdbManager::RefInfo> handlersInfo = getKVDBHandlersInfo();
+
+    // Create a temporal map with the reverse lookup indexed by Scope instead of Database.
+    std::map<std::string, kvdbManager::RefCounter> refCounterMap;
+
+    // Iterate over the map of DBs and scopes that are using them.
+    for (auto& dbEntry : handlersInfo)
+    {
+        auto dbName = dbEntry.first;
+        auto scopesUsingDB = dbEntry.second;
+
+        // Iterate over the scopes that are using thisDB.
+        for (auto& scopeEntry : scopesUsingDB)
+        {
+            std::string scopeName = scopeEntry.first;
```

Comment:
Please, remember to initialize the variables with explicitly declared type with the `{}`